test/nbrowser/DocUsageTracking.ts

```diff
@@ -0,0 +1,179 @@
+
+  describe('attachment usage without external attachments', function() {
+    let docId: string;
+
+    before(async () => {
+      docId = await session.tempNewDoc(cleanup, `AttachmentUsageTestDoc - internal`);
+    });
+
+    testAttachmentsUsage(() => docId);
+  });
+
+  describe('attachment usage with external attachments', function() {
+    let docId: string;
+
+    before(async () => {
+      docId = await session.tempNewDoc(cleanup, `AttachmentUsageTestDoc - external`);
+      const docApi = api.getDocAPI(docId);
+      await docApi.setAttachmentStore("external");
+      assert.equal((await docApi.getAttachmentStore()).type, "external");
+    });
+
+    testAttachmentsUsage(() => docId);
+  });
+});
+
+async function goToDocUsage() {
+  await driver.findWait('.test-tools-raw', 2000).click();
+
+  // Check that the Usage section exists.
+  await waitForDocUsage();
+}
+
+async function assertUsageMessage(text: string | null) {
+  if (text === null) {
+    assert.isFalse(await driver.find('.test-doc-usage-message').isPresent());
+  } else {
+    assert.equal(await driver.findWait('.test-doc-usage-message-text', 2000).getText(), text);
+  }
+}
+
+async function assertRowCount(currentValue: string, maximumValue?: string) {
+  await gu.waitToPass(async () => {
+    const rowUsage = await driver.find('.test-doc-usage-rows .test-doc-usage-value').getText();
+    const [, foundValue, foundMax] = rowUsage.match(/([0-9,]+) (?:of ([0-9,]+) )?rows/) || [];
+    assert.equal(foundValue, currentValue);
+    if (maximumValue) {
+      assert.equal(foundMax, maximumValue);
+    }
+  });
+}
+
+async function assertDataSize(currentValue: string, maximumValue?: string) {
+  await gu.waitToPass(async () => {
+    const dataUsage = await driver.find('.test-doc-usage-data-size .test-doc-usage-value').getText();
+    const [, foundValue, foundMax] = dataUsage.match(/([0-9,.]+) (?:of ([0-9,.]+) )?MB/) || [];
+    assert.equal(foundValue, currentValue);
+    if (maximumValue) {
+      assert.equal(foundMax, maximumValue);
+    }
+  });
+}
+
+async function assertAttachmentsSize(currentValue: string, maximumValue?: string) {
+  await gu.waitToPass(async () => {
+    const attachmentUsage = await driver.find('.test-doc-usage-attachments-size .test-doc-usage-value').getText();
+    const [, foundValue, foundMax] = attachmentUsage.match(/([0-9,.]+) (?:of ([0-9,.]+) )?GB/) || [];
+    assert.equal(foundValue, currentValue);
+    if (maximumValue) {
+      assert.equal(foundMax, maximumValue);
+    }
+  });
+}
+
+async function waitForDocUsage() {
+  await driver.findWait('.test-doc-usage-container', 8000);
+  await gu.waitToPass(async () => {
+    return assert.isFalse(await driver.find('.test-doc-usage-loading').isPresent());
+  });
+}
+
+async function addAttachmentColumn(columnName: string) {
+  await gu.toggleSidePanel('right', 'open');
+  await driver.find('.test-right-tab-field').click();
+  await gu.addColumn(columnName);
+  await gu.setType(/Attachment/);
+}
```

Comment:
This helper is not needed, there is a type parameter in the `addColumn`, which does the same thing as you.